mongo/mongo.go

```diff
@@ -132,6 +133,14 @@ func getEncoder(
 	return enc, nil
 }
 
+// newEncoderFn will return a function for constructing and encoder based on
+// the provided codec options.
+func newEncoderFn(opts *options.BSONOptions) codecutil.EncoderFn {
+	return func(w io.Writer) (*bson.Encoder, error) {
+		return getEncoder(w, opts, nil)
```

Comment:
Optional: You can reduce need to pass around a `Registry` if you incorporate it into the encoder here. Consider updating `newEncoderFn` to accept a `Registry` as well.

E.g.
```go
func newEncoderFn(opts *options.BSONOptions, reg *bsoncodec.Registry) codecutil.EncoderFn {
	return func(w io.Writer) (*bson.Encoder, error) {
		return getEncoder(w, opts, reg)
	}
}
```

Reply:
This is very clever, thank you!